scylla-gdb.py

```diff
@@ -2341,6 +2346,64 @@ def print_desc(seg_addr, desc):
                 addr += segment_size
 
 
+def shard_of(ptr):
+    return (int(ptr) >> 36) & 0xff
+
+class scylla_lsa_check(gdb.Command):
+    """
+    Runs consistency checks on the LSA state on current shard.
+    """
+    def __init__(self):
+        gdb.Command.__init__(self, 'scylla lsa-check', gdb.COMMAND_USER, gdb.COMPLETE_COMMAND)
+
+    def invoke(self, arg, from_tty):
+        segment_size = int(gdb.parse_and_eval('\'logalloc\'::segment::size'))
+
+        db = find_db()
+
+        # Collect segment* which are present in _segment_descs
+        in_buckets = set()
+        cache_region = lsa_region(db['_row_cache_tracker']['_region'])
+        shard = shard_of(cache_region.impl())
+        for bucket in std_array(cache_region.impl()['_segment_descs']['_buckets']):
+            for desc in intrusive_list(bucket):
+                if shard_of(int(desc.address)) != shard:
+                    gdb.write('ERROR: Out-of-shard segment: (logalloc::segment_descriptor*)0x%x, free_space=%d\n'
+                              % (int(desc.address), desc.free_space()))
```

Comment:
The more modern way of formatting strings is `"my string".format()`, the even more modern is `f"my string"` (although I couldn't always get this syntax to work.
We do use the `%` currently, so you don't have to change it but we should move to the `str.format()` eventually.